examples/IF_and_HH_responses_to_injection.py

```diff
@@ -0,0 +1,119 @@
+# encoding: utf-8
+"""
+Compare the response of IF and HH neurones when they recieve defferent type of current injections
+
+There are four "standard" current sources in PyNN which are shown here:
+
+    - DCSource
+    - ACSource
+    - StepCurrentSource
+    - NoisyCurrentSource
+
+Any other current waveforms can be implemented using StepCurrentSource.
+
+
+Usage: IF_and_HH_responses_to_injection.py [-h] [--plot-figure]  simulator
+
+positional arguments:
+  simulator      neuron, nest, brian or another backend simulator
+
+optional arguments:
+  -h, --help     show this help message and exit
+  --plot-figure  plot the simulation results to a file
+  --debug DEBUG  print debugging information
+
+"""
+
+
+# === Configure the simulator ================================================
+from pyNN.utility import get_simulator, normalized_filename
+sim, options = get_simulator(("--plot-figure", "Plot the simulation results to a file",
+                              {"action": "store_true"}))
+
+sim.setup()
+
+# === Create four cells of HH anf IF and inject the 4 types of currents =====================
+
+current_sources = [sim.DCSource(amplitude=1, start=20.0, stop=100.0),
+                   sim.StepCurrentSource(times=[20.0, 50.0, 100.0],
+                                         amplitudes=[0.4, 0.6, -0.2]),
+                   sim.ACSource(start=20.0, stop=100.0, amplitude=0.4,
+                                offset=0.1, frequency=10.0, phase=180.0),
+                   sim.NoisyCurrentSource(mean=0.5, stdev=0.2, start=20.0,
+                                          stop=100.0, dt=1.0)]
+
+hhcell1 = sim.Population(1, sim.HH_cond_exp())
```

Comment:
It would be better to create one population of size 4 rather than four populations each of size 1.